src/main/java/rx/observables/AbstractOnSubscribe.java

```diff
@@ -332,20 +332,22 @@ private SubscriptionProducer(SubscriptionState<T, S> state) {
         }
         @Override
         public void request(long n) {
-            if (n == Long.MAX_VALUE) {
-                for (; !state.subscriber.isUnsubscribed(); ) {
-                    if (!doNext()) {
-                        break;
-                    }
-                }
-            } else 
             if (n > 0 && state.requestCount.getAndAdd(n) == 0) {
```

Comment:
Thanks. I thought I added it because of the visibility changes, but it appears it got lost somewhere.